pkg/client/unversioned/policy.go

```diff
@@ -65,12 +65,10 @@ func setPolicyDefaults(config *restclient.Config) error {
 	if config.UserAgent == "" {
 		config.UserAgent = restclient.DefaultKubernetesUserAgent()
 	}
-	// TODO: Unconditionally set the config.Version, until we fix the config.
-	//if config.Version == "" {
-	copyGroupVersion := g.GroupVersion
-	config.GroupVersion = &copyGroupVersion
-	//}
-
+	if config.GroupVersion == nil || !isEnabledVersion(policy.GroupName, config.GroupVersion) {
```

Comment:
No, this is to enable defaults if passed versions do not match policy `GroupVersion`. Currently `restclient.Config` contains only single `GroupVersion`. Eg. I'm hitting `batch/v2alpha1` and I want client like that being created. But `batch/v2alpha1` is valid only for batch and only there should be enabled, all the remaining clients (apps, policy, etc.) should use their default versions. This is used then during caching clients in `pkg/kubectl/cmd/util/factory.go` I have a proper batchClient pointing to that version but others clients are still functional against their default groups. 

I think in the long run we will have to be able to pass multiple GV configuration in `restclient.Config`